src/components/views/rooms/RoomList.js

```diff
@@ -260,38 +264,41 @@ module.exports = React.createClass({
     }, 500),
 
     // Update which rooms and users should appear in RoomList as dictated by selected tags
-    updateSelectedTagsEntities: function() {
+    updateSelectedTagsRooms: function(dmRoomMap, updatedTags) {
         if (!this.mounted) return;
-        this._selectedTagsRoomIds = [];
-        this._selectedTagsUserIds = [];
-        FilterStore.getSelectedTags().forEach((tag) => {
-            this._selectedTagsRoomIds = this._selectedTagsRoomIds.concat(
-               this._groupStores[tag].getGroupRooms().map((room) => room.roomId),
-            );
+        updatedTags.forEach((tag) => {
```

Comment:
it feels odd that this function is doing half of its work based on the `updatedTags` that is passed in, and the other half based on `FilterStore.getSelectedTags()`. I would be inclined to split the function in two and call both halves from the FilterStore listener.

Reply:
But there isn't a case where you'd call one and not the other. Would it help if I passed the selected tags in as an argument?